src/fft128/mod.rs

```diff
@@ -0,0 +1,2204 @@
+        let mut twid_re1 = avec![0.0f64; n].into_boxed_slice();
+        let mut twid_im0 = avec![0.0f64; n].into_boxed_slice();
+        let mut twid_im1 = avec![0.0f64; n].into_boxed_slice();
+
+        init_negacyclic_twiddles(&mut twid_re0, &mut twid_re1, &mut twid_im0, &mut twid_im1);
+
+        Self {
+            twid_re0,
+            twid_re1,
+            twid_im0,
+            twid_im1,
+        }
+    }
+
+    /// Returns the vector size of the negacyclic FFT.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use concrete_fft::fft128::Plan;
+    /// let plan = Plan::new(32);
+    /// assert_eq!(plan.fft_size(), 32);
+    /// ```
+    pub fn fft_size(&self) -> usize {
+        self.twid_re0.len()
+    }
+
+    /// Performs a forward negacyclic FFT in place.
+    ///
+    /// # Note
+    ///
+    /// The values in `buf_re0`, `buf_re1`, `buf_im0`, `buf_im1` must be in standard order prior to
+    /// calling this function. When this function returns, the values in `buf_re0`, `buf_re1`, `buf_im0`, `buf_im1` will contain the
+    /// terms of the forward transform in bit-reversed order.
+    #[track_caller]
+    pub fn fwd(
+        &self,
+        buf_re0: &mut [f64],
+        buf_re1: &mut [f64],
+        buf_im0: &mut [f64],
+        buf_im1: &mut [f64],
+    ) {
+        assert_eq!(buf_re0.len(), self.fft_size());
+        assert_eq!(buf_re1.len(), self.fft_size());
+        assert_eq!(buf_im0.len(), self.fft_size());
+        assert_eq!(buf_im1.len(), self.fft_size());
+
+        negacyclic_fwd_fft(
+            buf_re0,
+            buf_re1,
+            buf_im0,
+            buf_im1,
+            &self.twid_re0,
+            &self.twid_re1,
+            &self.twid_im0,
+            &self.twid_im1,
+        );
+    }
+
+    /// Performs an inverse negacyclic FFT in place.
+    ///
+    /// # Note
+    ///
+    /// The values in `buf_re0`, `buf_re1`, `buf_im0`, `buf_im1` must be in bit-reversed order
+    /// prior to calling this function. When this function returns, the values in `buf_re0`,
+    /// `buf_re1`, `buf_im0`, `buf_im1` will contain the terms of the inverse transform in standard
+    /// order.
+    #[track_caller]
+    pub fn inv(
+        &self,
+        buf_re0: &mut [f64],
+        buf_re1: &mut [f64],
+        buf_im0: &mut [f64],
+        buf_im1: &mut [f64],
+    ) {
+        assert_eq!(buf_re0.len(), self.fft_size());
+        assert_eq!(buf_re1.len(), self.fft_size());
+        assert_eq!(buf_im0.len(), self.fft_size());
+        assert_eq!(buf_im1.len(), self.fft_size());
+
+        negacyclic_inv_fft(
+            buf_re0,
+            buf_re1,
+            buf_im0,
+            buf_im1,
+            &self.twid_re0,
+            &self.twid_re1,
+            &self.twid_im0,
+            &self.twid_im1,
+        );
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use alloc::vec;
+    use rand::random;
+
+    extern crate alloc;
+
+    #[test]
+    fn test_wrapper() {
+        let n = 1024;
```

Comment:
I think you tested more n values for the f64 fft right ? is it possible to do the same here ?

Reply:
yeah, i could test all the sizes up to some reasonable limit